autotest.pm

```diff
@@ -345,6 +350,44 @@ sub loadtestdir {
     }
 }
 
+sub parse_serial_output {
+    return -1 if !-e 'serial0';
+
+    my ($serial_file_pos, $t) = @_;
+    my $failures = $t->{serial_failures};
+    
+    open my $ifh, '<', 'serial0' or die "Unable to open file: $!";
+    seek $ifh, $serial_file_pos, 0;
+    my $die = 0;
+    while (defined(my $line = <$ifh>)) {
+        chomp $line;
+        for my $type (keys %$failures) {
+            for my $regexp (@{$failures->{$type}}) {
+                my $re = quotemeta $regexp;
+                if ($line =~ /$regexp/ || $line =~ /$re/) {
```

Comment:
The user should store the regex in the hash with `qr/../` so they are compiled once

Reply:
@richiejp 

Compiling regexp doesn't mean it's faster and **not** using qr doesn't mean multiple regexp compilations . Using qr **might be faster** if we have variables interpolated. If there's no variables to be interpolated (which i assume is going to be most of the cases - since we will be looking for specific strings) the regular expression is compiled once and cached.

Reply:
I would suggest to expect the user to write the expressions with qr/exp/ already, because of compiling only once, and because you should not try to cover for an user, giving another point of possible failure by adding quotemeta. Unless you intention is giving the possibility of having support for both cases... Which imo, not the best idea